app/models/annotation.rb

```diff
@@ -0,0 +1,15 @@
+class Annotation < ApplicationRecord
+  has_one :map_overlay
+  has_one :building
+  validates_presence_of :annotation_text
+
+  def annotation
+    annotation_text
+  end
+
+  # seems clunky
```

Comment:
Yes this is a good job for delegation (such as `delegate :name, to: :map_layer, prefix: true`)